src/zappy_gui_src/Core/Network/pack.c

```diff
@@ -0,0 +1,132 @@
+/*
+** EPITECH PROJECT, 2022
+** ZAPPYMIRROR
+** File description:
+** unpack
+*/
+
+/// \file src/zappy_gui_src/Core/Network/pack.c
+
+#include "entity.h"
+#include "entity_types.h"
+#include "container.h"
+#include "player.h"
+#include "tile.h"
+#include "eggs.h"
+#include <string.h>
+#include <sys/queue.h>
+#include <stdio.h>
+
+char *pack_int(int val)
+{
+    char *tmp;
+    char str[80];
+
+    sprintf(str, "%d", val);
+    tmp = (char*)malloc(sizeof(char) * strlen(str) + 2);
+    strcat(tmp, str);
+    strcat(tmp, ";");
+    return tmp;
+}
+
+char *pack_str(char *val)
+{
+    char *tmp;
+
+    tmp = (char*)malloc(sizeof(char) * strlen(val) + 2);
+    strcat(tmp, val);
+    strcat(tmp, ";");
+    return tmp;
+}
+
+char *pack_container(container_t *cont)
+{
+    char *tmp;
+    char str[80];
+
+    tmp = strdup(";inventory{");
```

Comment:
The size of `tmp` won't be the good size with strdup if you want to use strcat after. Calculate the total length of the string before to add value in it.

Reply:
+1